smallLibraries/Google.Cloud.Templates/content/gcpmvc/CSharp/Program.cs

```diff
@@ -0,0 +1,201 @@
+                .UseStartup<Startup>()
+                .Build();
+#elif (Framework2)
+            WebHost.CreateDefaultBuilder(args)
+                .UseContentRoot(Directory.GetCurrentDirectory())
+                .UseIISIntegration()
+                .ConfigureAppConfiguration((context, configBuilder) => {
+                    HostingEnvironment = context.HostingEnvironment;
+
+                    configBuilder.SetBasePath(HostingEnvironment.ContentRootPath)
+                        .AddJsonFile("appsettings.json", optional: true, reloadOnChange: true)
+                        .AddJsonFile($"appsettings.{HostingEnvironment.EnvironmentName}.json", optional: true)
+                        .AddEnvironmentVariables();
+
+                    Configuration = configBuilder.Build();
+                    GcpProjectId = GetProjectId(Configuration);
+                })
+                .ConfigureServices(services => {
+                    // Add framework services.Microsoft.VisualStudio.ExtensionManager.ExtensionManagerService
+                    services.AddMvc();
+
+                    if (HasGcpProjectId)
+                    {
+                        // Enables Stackdriver Trace.
+                        services.AddGoogleTrace(options => options.ProjectId = GcpProjectId);
+                        // Sends Exceptions to Stackdriver Error Reporting.
+                        services.AddGoogleExceptionLogging(
+                            options => {
+                                options.ProjectId = GcpProjectId;
+                                options.ServiceName = GetServiceName(Configuration);
+                                options.Version = GetVersion(Configuration);
+                            });
+                        services.AddSingleton<ILoggerProvider>(sp => GoogleLoggerProvider.Create(GcpProjectId));
+                    }
+                })
+                .ConfigureLogging(loggingBuilder => {
+                    loggingBuilder.AddConfiguration(Configuration.GetSection("Logging"));
+                    if (HostingEnvironment.IsDevelopment())
+                    {
+                        // Only use Console and Debug logging during development.
+                        loggingBuilder.AddConsole(options =>
+                            options.IncludeScopes = Configuration.GetValue<bool>("Logging:IncludeScopes"));
+                        loggingBuilder.AddDebug();
+                    }
+                })
+                .Configure((app) => {
+                    var logger = app.ApplicationServices.GetService<ILoggerFactory>().CreateLogger("Startup");
+                    if (HasGcpProjectId)
+                    {
+                        // Sends logs to Stackdriver Error Reporting.
+                        app.UseGoogleExceptionLogging();
+                        // Sends logs to Stackdriver Trace.
+                        app.UseGoogleTrace();
+
+                        logger.LogInformation(
```

Comment:
Can we unwrap all of these lines? I know they'll be fairly long, but I think they're simple enough to read.